src/python/zquantum/core/cost_function.py

```diff
@@ -31,11 +31,26 @@
 )
 from .utils import ValueEstimate, create_symbols_map
 
+import re
+
+
+def check_for_digit_in_string(text):
+    return int(text) if text.isdigit() else text
+
+
+def natural_keys(
```

Comment:
Would it be possible to add tests with those examples from our conversation on Slack?